### PR TITLE
In Real Life - AWS: IAM

### PR DESCRIPTION
**Pull request by matthewreed26**

### CODE REVIEW

File: posts/irl-aws-iam.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+- AWS
+- IAM
+date: 2020-12-04T00:00:00.000Z
+title: "In Real Life - Amazon Web Services: Identity Access Management"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2020/12/iam_kitten.png
+---
+
+## A Quick Note about In Real Life
+
+For some reason, non-physical concepts have always confused a lot of people unfamiliar with the digital realm. Even those who do work in the IT-services industry can get tripped up by subtle trivialities. This is an attempt to distill the complexities down into more digestible ideas for everyone.
```
Comment (yhippa) [inline posts/irl-aws-iam.md:14]: Ironically, I think this part is too verbose. I think you can use less complex words and dumb it down a bit to make the reader more comfortable and less scared about what you're about to dive into.

File: posts/irl-aws-iam.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+## Identity Access Management a.k.a. Sharing is Caring
+
+To begin, someone who has never heard of "Identity Access Management" (or IAM for short) might be unsure about what it should do in the first place. The questions to ask here are:
```
Comment (yhippa) [inline posts/irl-aws-iam.md:18]:
```suggestion
To begin, someone who has never heard of [Identity Access Management (IAM)](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html) might be unsure about what it should do in the first place. The questions to ask here are:
```

File: posts/irl-aws-iam.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+1. What is being worked on?
+1. Who is working on it?
```
Comment (yhippa) [inline posts/irl-aws-iam.md:21]: I'm not sure what you mean by these questions and how they're relevant to the reader at this point. Can you elaborate on what you're trying to get at?

File: posts/irl-aws-iam.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+### Letting Others In
+
+Sharing can be a hard concept to grasp. Many questions exist not just about how much to share but also with whom and for how long. This is why permissions are given to guests entering a home for a dinner party, children when they want to stay up past their bedtime on New Year's Eve, or soldiers when they need to speak freely to their superior. This system of people and permissions is intuitively understood.
+
+But when translating this to the digital space, it is not as straightforward. AWS attempts to establish a sharing system by defining 4 key terms:
+
+1. Users - people
+1. Groups - a collection of users
+1. Policies - a contract of permissions
+1. Roles - entities assumed to operate under certain permissions
```
Comment (yhippa) [inline posts/irl-aws-iam.md:40]: You probably want to elaborate on these.

File: posts/irl-aws-iam.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+Here, what is best to keep in mind is Policies are paramount. They delineate the permitted actions a service can take on one or more resources (i.e. A maid service may enter the home). Everything else cascades from these Policies.
```
Comment (yhippa) [inline posts/irl-aws-iam.md:42]: What is "everything else" here?

File: posts/irl-aws-iam.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+Groups have Policies that they must adhere to. These could be organizations or teams, but everyone is designated to act under the same Policy restrictions. These Users are identified by a name and given a password. It is a best practice to assign Users to a Group so that they only have certain job functions.
+
+It is worth specifying that Roles are **not** typically Users in AWS. Instead, think of these entities like robots filling in for actors in a Broadway play. As robots, they do not improvise. They are permitted to perform only the set of directions they are told so that the play can proceed. In this way, they adhere to Policies similar to Groups.
+
+### Locks, Keys, and Secrets
+
+Just as much as IAM is about letting others in, it is equally about keeping others out. It was previously mentioned that users authenticate themselves in AWS by a name and password. This is because they (and only they) should be trusted to access the services/resources outlined in their group's policies. Anything that is not in these policies should be off-limits.
```
Comment (yhippa) [inline posts/irl-aws-iam.md:50]: Don't use passive voice.

File: posts/irl-aws-iam.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+Think of it like a bank vault with smaller vaults locked away inside. The account owner should ultimately only have access to their money. But the bank's teller needs access to the main and underlying vaults in order to withdraw/deposit the funds for any person's account. Both are authenticated in the system with different levels of authorization.
```
Comment (yhippa) [inline posts/irl-aws-iam.md:52]: Is there a simpler, more relatable example you can use here?

File: posts/irl-aws-iam.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+### Flip it and Reverse It
```
Comment (yhippa) [inline posts/irl-aws-iam.md:54]: Avoid idioms.

**Review by yhippa — COMMENTED** [review]
I really like the cat sitting analogy. One thing I noticed is that there's a lot of superfluous language used everywhere that's a little distracting and fatiguing. Wherever possible, avoid complex grammatical phrases and favor short, understandable, and clear sentences.

**Comment by matthewreed26** [comment]
> I really like the cat sitting analogy. One thing I noticed is that there's a lot of superfluous language used everywhere that's a little distracting and fatiguing. Wherever possible, avoid complex grammatical phrases and favor short, understandable, and clear sentences.

You know, you are so right. I fluff at times and it distracts from the intended simplification. Thank you for the feedback!

**Review by pkrish123 — APPROVED** [review]

**Review by yhippa — APPROVED** [review]
